.mocharc.js

```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  require: ['source-map-support/register', 'test/support/modules_helper.js', 'test/support/test_helper.js'],
```

Comment:
good point! these hooks weren't being run in the browser at all so i made a commit to move them into shared_helper.js in another branch (https://github.com/ably/ably-js/pull/1645/commits/359c5143c169839594b22f68a015a227beb64184), i might just rebase this PR onto that one for the sake of simplicity rather than duplicating the commit

Reply:
cool, yeah i'm gonna wait for #1645 to land on the integration branch and rebase this PR (which will resolve this comment) so we can hold off merging until then

Reply:
Oh, I see!
I've noticed those new `afterEach` hooks in shared_helper and was wondering if that was a replacement for root_hooks.js, and turns out it is.

Now, I believe root_hooks.js was never registered in browser tests, because even though it is listed in `browser_file_list.js`, in `browser_setup.js` there are TEST_REGEXP and TEAR_DOWN_REGEXP regexps that filter out anything that is not `*.test.js` or `*tear_down.js` -> `root_hooks.js` doesn't match. And there is no other place where `root_hooks.js` would be imported in browser setup, so I guess that was the culprit.

I feel like separate `root_hooks.js` file was a better place for hooks like these, they seem a bit out of place for `shared_helper.js`.

That being said, current browser test setup is a maze and I don't want to drag on this PR for much longer. I believe Unified Test Suite will be the opportunity for us to re-implement browser tests with more modern setup.
So we can leave this as it is currently in this PR